tests/unit/test_index_html_page.py

```diff
@@ -118,6 +118,29 @@ def test_get_html_response_no_head(url):
     ]
 
 
+@mock.patch("pip._internal.index.logger")
+def test_get_html_response_dont_log_clear_text_password(m_logger):
```

Comment:
You shouldn't need to mock logging. I'll repeat my suggestion [from before](https://github.com/pypa/pip/pull/6295#discussion_r259642280):
 >You can look at `test_get_legacy_build_wheel_path__no_names()` and nearby functions for examples of using the `caplog` fixture (since the code wouldn't be running in a separate subprocess).

Reply:
I forgot about it, and fixed in the next commit